crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -697,6 +697,37 @@ impl<'a> MoveTestAdapter<'a> for SuiTestAdapter<'a> {
                 })?;
 
                 let state = self.compiled_state();
+                let original_named_address_mapping = state.named_address_mapping.clone();
```

Comment:
The change that you're making to the mapping is quite easy to invert (on the way in you want to remove the entry for the original package and add an entry for the upgrade package, and on the way out you can do the opposite).  It's probably better to do it that way rather than copy the whole map to preserve its state.

Reply:
what I sense blocked me from doing this is that I made mutation conditional on `dependencies.is_empty`, but with the planned changes this should simplify and work the way you describe